### PR TITLE
Espresso test (with structure change)

### PR DESCRIPTION
**Pull request by hannesa2**
This run Espresso tests and make screenshots on each test.

Most CI development can only be tested on CI, that's why development was made here hannesa2#6 to avoid a lot of noise here.

### CODE REVIEW

**Comment by hannesa2** [comment]
It is based on #190 to make it work. 
image

Without #190 it's not able to start tests, please see #188

**Comment by hannesa2** [comment]
Now it runs for most Android projects smoke tests, and archive the test results **and a screenshot**

image

File: .github/workflows/Android-CI.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ on: [pull_request]
 jobs:
   buildTest:
     name: Build & Test JDK ${{ matrix.java_version }}
-    runs-on: ubuntu-latest
+    runs-on: macOS-latest
```
Comment (topherbuckley) [inline .github/workflows/Android-CI.yml:8]: Whats the reason for the OS change?

Comment (hannesa2) [reply .github/workflows/Android-CI.yml:8]: macOS comes with hardware acceleration to run Emulator with x86 fast

**Comment by topherbuckley** [comment]
Wow you've done a lot on this! Nice! I'll take a closer look at this and the other CI stuff tomorrow.

File: .github/workflows/Android-CI.yml (hunk 1 above)
Comment (topherbuckley) [reply .github/workflows/Android-CI.yml:8]: Where'd you see that? I didn't see anything mentioned in the [docs](https://docs.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners). Cool if it actually runs faster though!

Comment (hannesa2) [reply .github/workflows/Android-CI.yml:8]: I can't tell you exactly, I simply know it. You can test it and change to `ubuntuLatest`, then you see the result

**Pull request merged**